deployment/cdk/opensearch-service-migration/lib/historical-capture-stack.ts

```diff
@@ -1,62 +1,97 @@
+        dpPipelineData = dpPipelineData.replace("<TARGET_CLUSTER_HOST>", props.targetEndpoint);
+        // Base64 encode
+        let encodedPipeline = Buffer.from(dpPipelineData).toString("base64");
+
+        // Create secret using Secrets Manager
+        const dpPipelineConfigSecret = new SMSecret(this, "dpPipelineConfigSecret", {
+            secretName: `${historicalCaptureFargateTask.family}-${historicalCaptureContainer.containerName}-pipelineConfig`,
+            secretStringValue: SecretValue.unsafePlainText(encodedPipeline)
         });
+        // Add secret to container
+        historicalCaptureContainer.addSecret("INLINE_PIPELINE", 
+            ECSSecret.fromSecretsManager(dpPipelineConfigSecret)
+        );
 
-        // Create Fargate Service
-        const historicalCaptureFargateService = new FargateService(this, "historicalCaptureFargateService", {
-            cluster: ecsCluster,
-            taskDefinition: historicalCaptureFargateTask,
-            desiredCount: 1
+        // The Migration Domain SG disallows all outgoing, but this prevents the ECS run task
+        // command from pulling the Docker image from ECR. Thus, create an SG to allow this.
+        // TODO - Limit outbound to only ECR endpoints somehow
+        const fetchMigrationSecurityGroup = new SecurityGroup(this, 'fetchMigrationECRAccessSG', {
+            vpc: ecsCluster.vpc
         });
 
+        // CFN outputs that will be consumed by Copilot
+        new CfnOutput(this, 'FMTaskDefArn', {
+            exportName: 'FetchMigrationTaskDefinitionArn',
+            value: historicalCaptureFargateTask.taskDefinitionArn,
+            description: 'ARN of the ECS Task Definition for Fetch Migration',
+        });
+        new CfnOutput(this, 'FMTaskRoleArn', {
+            exportName: 'FetchMigrationTaskRoleArn',
+            value: historicalCaptureFargateTask.taskRole.roleArn,
+            description: 'ARN of the Task Role for Fetch Migration',
+        });
+        new CfnOutput(this, 'FMExecRoleArn', {
+            exportName: 'FetchMigrationExecutionRoleArn',
+            value: historicalCaptureFargateTask.obtainExecutionRole().roleArn,
+            description: 'ARN of the Execution Role for Fetch Migration',
+        });
+        // Documentation - https://docs.aws.amazon.com/cli/latest/reference/ecs/run-task.html#options
+        let networkConfigJson = {
+            "awsvpcConfiguration": {
+                "subnets": ecsCluster.vpc.privateSubnets.map(_ => _.subnetId),
+                "securityGroups": [process.env.MIGRATION_DOMAIN_SG_ID, fetchMigrationSecurityGroup.securityGroupId]
+            }
+        }
+        let networkConfigString = JSON.stringify(networkConfigJson)
+        // Output the ECS run task command using template literals
+        let executionCommand = `aws ecs run-task --task-definition ${historicalCaptureFargateTask.taskDefinitionArn}`
```

Comment:
Where should this command be run from? I'm trying to put together how this plays with the container you created

Reply:
The Fetch Migration container will host the task that this command launches. The user is expected to run this command on the Migration Console to kick off Fetch Migration.

Reply:
Ah okay so launching this task won't spin up another container it will use the existing container in the cluster

Reply:
That's correct. Note that it will launch a separate task in the historical migration cluster though - there's currently no protection from having multiple tasks performing duplicate work